sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -181,6 +182,11 @@ case class CreateViewCommand(
    * Permanent views are not allowed to reference temp objects, including temp function and views
    */
   private def verifyTemporaryObjectsNotExists(sparkSession: SparkSession): Unit = {
+    val lookup = new LookupCatalog {
+      override protected def lookupCatalog(name: String): CatalogPlugin = sparkSession.catalog(name)
+    }
+    import lookup._
```

Comment:
Do we need to build a new `LookupCatalog`?

How about directly using the analyzer's AsTableIdentifier?
> sparkSession.sessionState.analyzer.AsTableIdentifier(ident))

Reply:
Good idea!

Reply:
I agree, ideally we could use the analyzer for this purpose.